src/DataObject/Controller/Search/GetConfigurationController.php

```diff
@@ -0,0 +1,86 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Bundle\StudioBackendBundle\DataObject\Controller\Search;
+
+use OpenApi\Attributes\Get;
+use OpenApi\Attributes\JsonContent;
+use Pimcore\Bundle\StudioBackendBundle\Controller\AbstractApiController;
+use Pimcore\Bundle\StudioBackendBundle\Exception\Api\NotFoundException;
+use Pimcore\Bundle\StudioBackendBundle\Grid\Schema\DetailedConfiguration;
+use Pimcore\Bundle\StudioBackendBundle\Grid\Service\ConfigurationServiceInterface;
+use Pimcore\Bundle\StudioBackendBundle\OpenApi\Attribute\Parameter\Path\StringParameter;
+use Pimcore\Bundle\StudioBackendBundle\OpenApi\Attribute\Response\DefaultResponses;
+use Pimcore\Bundle\StudioBackendBundle\OpenApi\Attribute\Response\SuccessResponse;
+use Pimcore\Bundle\StudioBackendBundle\OpenApi\Config\Tags;
+use Pimcore\Bundle\StudioBackendBundle\Util\Constant\HttpResponseCodes;
+use Pimcore\Bundle\StudioBackendBundle\Util\Constant\UserPermissions;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\Routing\Attribute\Route;
+use Symfony\Component\Security\Http\Attribute\IsGranted;
+use Symfony\Component\Serializer\SerializerInterface;
+
+/**
+ * @internal
+ */
+final class GetConfigurationController extends AbstractApiController
+{
+    public function __construct(
+        SerializerInterface $serializer,
+        private readonly ConfigurationServiceInterface $gridConfigurationService,
+    ) {
+        parent::__construct($serializer);
+    }
+
+    /**
+     * @throws NotFoundException
+     */
+    #[Route(
+        '/data-object/search/configuration/{classId?}',
```

Comment:
```suggestion
        '/data-object/search/configuration/{classId}',
```
should it be like this?

Reply:
Nop.. classId has to be optional, that's why there is a `?`

Reply:
Ah I see ... but the does it not need to be the same here?
https://github.com/pimcore/studio-backend-bundle/pull/911/files#diff-91c31968b6e99f9f0907012a0e6020afd877a5d981f7563472140d4a14b9e837R59

Reply:
This does not work, Swagger can not work with `?`
It is defined here, that it is optional
https://github.com/pimcore/studio-backend-bundle/pull/911/files#diff-91c31968b6e99f9f0907012a0e6020afd877a5d981f7563472140d4a14b9e837R69

Reply:
Alright, thanks for the clarification! :)